classes/editor_framework.php

```diff
@@ -0,0 +1,178 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * \mod_hvp\framework class
+ *
+ * @package    mod_hvp
+ * @copyright  2016 Joubel AS
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_hvp;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once __DIR__ . '/../autoloader.php';
+
+/**
+ * Moodle's implementation of the H5P Editor framework interface.
+ *
+ * @package    mod_hvp
+ * @copyright  2016 Joubel AS
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class editor_framework implements \H5peditorStorage {
+
+    /**
+     * Empty contructor.
+     */
+    function __construct() { }
+
+    /**
+     * Implements getLanguage().
+     */
+    public function getLanguage($name, $major, $minor, $lang) {
+        global $DB;
+
+        return $DB->get_field_sql(
+            "SELECT hlt.language_json
+               FROM {hvp_libraries_languages} hlt
+               JOIN {hvp_libraries} hl ON hl.id = hlt.library_id
+              WHERE hl.machine_name = ?
+                AND hl.major_version = ?
+                AND hl.minor_version = ?
+                AND hlt.language_code = ?
+            ", array(
+                $name,
+                $major,
+                $minor,
+                $lang
+            )
+        );
+    }
+
+    /**
+     * Implements addTmpFile().
+     */
+    public function addTmpFile($file) {
+        // TODO: Keep track of tmp files.
+    }
+
+    /**
+     * Implements keepFile().
+     */
+    public function keepFile($oldpath, $newpath) {
+        // TODO: No longer a tmp file.
+    }
+
+    /**
+     * Implements removeFile().
+     */
+    public function removeFile($path) {
+        // TODO: Removed from file tracking.
+    }
+
+    /**
+     * Implements getLibraries().
```

Comment:
Also it's difficult to figure out what exactly the function is doing unless you know beforehand how H5P works . So comments meant for H5P novices would be helpful.